dotnet-client-libraries/YamlDotNet/App/UseCases/DeserializerValidation.cs

```diff
@@ -0,0 +1,22 @@
+using System.ComponentModel.DataAnnotations;
+using YamlDotNet.Core;
+using YamlDotNet.Serialization;
+using static System.Environment;
+
+namespace App.UseCases;
+
+public class DeserializerValidation(INodeDeserializer nodeDeserializer) : INodeDeserializer
+{
+    public bool Deserialize(IParser reader, Type expectedType, Func<IParser, Type, object?> nestedObjectDeserializer,
+        out object? value)
+    {
+        if (!nodeDeserializer.Deserialize(reader, expectedType, nestedObjectDeserializer, out value)) return false;
+
+        var context = new ValidationContext(value);
+        var results = new List<ValidationResult>();
+        if (Validator.TryValidateObject(value, context, results, true)) return true;
```

Comment:
Any of these `if` conditions that have the `return` statement on the same line should be split into two.

```suggestion
        if (Validator.TryValidateObject(value, context, results, true))
            return true;
```

I see the single-line format used in some of the microsoft docs and elsewhere, but on here having two lines is preferred.